addons/sale_project/models/project_milestone.py

```diff
@@ -7,18 +7,40 @@ class ProjectMilestone(models.Model):
     _name = 'project.milestone'
     _inherit = 'project.milestone'
 
+    def _default_sale_line_id(self):
+        project_id = self._context.get('default_project_id')
+        if not project_id:
+            return []
+        project = self.env['project.project'].browse(project_id)
+        return project.sale_line_id.filtered(lambda sol: sol.qty_delivered_method == 'milestones')
```

Comment:
I discussed with Alexandra about that and we decided to select the first SOL of the SO linked to the project for which the product is a milestone.

And so we should have something like that:
```suggestion
        return self.env['sale.order.line'].search([
            ('order_id', '=', project.sale_order_id.id), 
            ('qty_delivered_method', '=', 'milestones'),
        ], limit=1)
```
(I did not test this code. 🙂)